duos_ui/lib/screens/chat_page.dart

```diff
@@ -1,201 +1,195 @@
-    return MaterialApp(
-      home: Scaffold(
-        appBar: AppBar(
-            backgroundColor: Colors.transparent,
-            centerTitle: true,
-            title: Text(
-              personName.value,
-              style: const TextStyle(color: Colors.black),
-            ),
-            leading: const Material(
-              child: IconButton(
-                onPressed: (null),
-                icon: Icon(Icons.arrow_back),
-              ),
-            ),
-            actions: [
-              Padding(
-                padding: const EdgeInsets.only(right: 24),
-                child: Avatar.small(url: "https://picsum.photos/200/300"),
-              ),
-            ],
-            shape: Border(bottom: BorderSide(color: Colors.black, width: 4)),
-            elevation: 0),
-        body: Column(
-          children: [Expanded(child: DemoMessage()), ActionBar()],
-        ),
-      ),
-    );
-  }
+  ChatPageArguments(
+      {required this.peerUid, required this.peerName, required this.peerImg});
 }
 
-class DemoMessage extends StatelessWidget {
-  const DemoMessage({Key? key}) : super(key: key);
+class ChatPage extends StatefulWidget {
+  const ChatPage({Key? key, required this.arguments}) : super(key: key);
+
+  final ChatPageArguments arguments;
 
   @override
-  Widget build(BuildContext context) {
-    return ListView(
-      children: [
-        _DateLabel(label: "Yesterday"),
-        _MessageTile(message: "Hey how is it going"),
-        OwnMessageTile(message: "how is your day"),
-        _MessageTile(message: "it is good"),
-        _MessageTile(message: "how about you"),
-        OwnMessageTile(message: "it could be better"),
-      ],
-    );
-  }
+  ChatPageState createState() => ChatPageState();
 }
 
-class _DateLabel extends StatelessWidget {
-  const _DateLabel({
-    Key? key,
-    required this.label,
-  }) : super(key: key);
+class ChatPageState extends State<ChatPage> {
+  late String uid1;
+  late String uid2;
+  List<QueryDocumentSnapshot> listMessage = [];
+  int limit = 25;
 
-  final String label;
+  TextEditingController textEditingController = TextEditingController();
+  ScrollController listScrollController = ScrollController();
+
+  late ChatProvider chatProvider;
+  late AuthProvider authProvider;
 
   @override
-  Widget build(BuildContext context) {
-    return Center(
-      child: Padding(
-        padding: const EdgeInsets.symmetric(vertical: 32.0),
-        child: Text(label),
-      ),
-    );
+  void initState() {
+    super.initState();
+    chatProvider = context.read<ChatProvider>();
+    authProvider = context.read<AuthProvider>();
+
+    readLocal();
   }
-}
 
-class _MessageTile extends StatelessWidget {
-  const _MessageTile({
-    Key? key,
-    required this.message,
-  }) : super(key: key);
 
-  final String message;
+  void readLocal() {
+    // uid1 = authProvider.sub;
+    // uid2 = widget.arguments.peerUid;
+    uid1 = "PbsHaVsmcZQtspSJvAPlzgCPmP72";
```

Comment:
hardcoded value?

Reply:
for its hardcoded. will be changed once i get the contact page working